### PR TITLE
Feature error patterns

### PR DESCRIPTION
**Pull request by julian-berbel**

### CODE REVIEW

File: lib/mumukit/error_pattern.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class ErrorPattern
```
Comment (flbulgarelli) [inline lib/mumukit/error_pattern.rb]: Please put this class on the `Mumukit` namespace

File: lib/mumukit/templates/query_hook.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module Mumukit::Templates
+  class QueryHook < FileHook
+    def post_process_file(_file, result, status)
+      error_patterns.each { |it| return it.transform(result, status) if it.matches? result } if status.failed?
```
Comment (flbulgarelli) [inline lib/mumukit/templates/query_hook.rb]: Could we extract this code to a mixin? I think there implementors of QueryHook that inherit from a different class than FileHook. 

I would like to be able to introduce this new behavior without requiring modifications to the inheritance chain

Comment (luchotc) [reply lib/mumukit/templates/query_hook.rb]: Can't this: `it.transform(result, status) if it.matches? result` be a method of ErrorPattern?

File: lib/mumukit/templates.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@ module Templates
 require_relative './templates/with_cookie'
 
 require_relative './templates/file_hook'
+require_relative './templates/query_hook'
```
Comment (flbulgarelli) [inline lib/mumukit/templates.rb]: The filename is wrong

**Pull request merged**